src/hotspot/share/opto/node.hpp

```diff
@@ -1627,10 +1641,35 @@ class Unique_Node_List : public Node_List {
     Node_List::clear();
     _clock_index = 0;
   }
+  void ensure_empty() {
+    assert(size() == 0, "must be empty");
+    clear(); // just in case
+  }
 
   // Used after parsing to remove useless nodes before Iterative GVN
   void remove_useless_nodes(VectorSet& useful);
 
+  // If the idx of the Nodes change, we must recompute the VectorSet
+  void recompute_idx_set() {
+    _in_worklist.clear();
+    for (uint i = 0; i < size(); i++) {
+      Node* n = at(i);
+      _in_worklist.set(n->_idx);
+    }
+  }
+
+#ifndef PRODUCT
```

Comment:
```suggestion
#ifdef ASSERT
```

We don't need this in the optimized build.